src/test/ui/pattern/usefulness/match-non-exhaustive.stderr

```diff
@@ -6,11 +6,11 @@ LL |     match 0 { 1 => () }
    |
    = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
 
-error[E0004]: non-exhaustive patterns: `_` not covered
+error[E0004]: non-exhaustive patterns: `std::i32::MIN..=std::i32::MAX` not covered
```

Comment:
`_` was better here.

Reply:
This looks like an improvement to me. `MIN..=MAX` tells me what the type is, which I might not know from the scrutinee itself or from the patterns (suppose it's a constant). It also tells me what the possible values are. Meanwhile, `_` only helps if `_ => ...` is really what I want.

A note might be useful tho, something like:

> = note: the whole range is uncovered

Reply:
I like the note suggestion. How should the decision be made in a case like this ?

Reply:
We let @estebank decide diagnostics stuff? :P

Reply:
I decided to go with @varkor's preference because the note suggestion was (imho) too ugly to code. We can always change that in a subsequent PR if we really want

Reply:
I'm ok with either way. That being said, this is enough of a corner case that it could be handled specially, by detecting it is `_` and providing something like

```
error[E0004]: non-exhaustive patterns: entire range of `i32` is not covered
```

As for using a `note`, that's a good solution for cases like this.

With all of that, I'm ok with handling this in a follow up PR. @Nadrieril would you mind filing a papercut ticket for this after this PR merges?